src/Components/Components/src/Profiling/ComponentsProfiling.cs

```diff
@@ -13,14 +13,9 @@ internal abstract class ComponentsProfiling
         // is so that if we later have two different implementations (one for WebAssembly, one for
         // Server), the execution characteristics of calling Start/End will be unchanged and historical
         // perf data will still be comparable to newer data.
-        public static readonly ComponentsProfiling Instance;
-
-        static ComponentsProfiling()
-        {
-            Instance = RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER"))
-                ? new WebAssemblyComponentsProfiling()
-                : (ComponentsProfiling)new NoOpComponentsProfiling();
-        }
+        public static readonly ComponentsProfiling Instance = PlatformInfo.IsWebAssembly
+            ? new WebAssemblyComponentsProfiling()
+            : (ComponentsProfiling)new NoOpComponentsProfiling();
```

Comment:
Is the cast needed here? NoOpComponentsProfiling already inherits ComponentsProfiling. So, to me it seems to be valid to assign an object of a sub-type to a field of the base type.

Reply:
I think https://github.com/dotnet/csharplang/issues/2460 should fix it in C# 9.0.